docs/pci/contribution-guidelines.md

```diff
@@ -0,0 +1,219 @@
+* device hot-plugging through PCIe Hotplug
+* pass-through of physical devices, like GPUs or EBS volumes, through VFIO.
+
+## Challenges
+
+Supporting PCIe in Firecracker and, in particular, device pass-through, introduces new challenges. Namely:
+
+* **overheads:** supporting the full PCI specification might negatively impact the boot time and 
+  memory overheads of Firecracker VMs. 
+    * We can mitigate this by allowing for completely disabling PCIe support via VM configuration 
+      when more lightweight virtualization is preferred.
+* **oversubscription:** simple PCIe device passthrough using VFIO requires the VMM to allocate the 
+  entire physical memory of the VM to allow for DMA from the device.
+    * Solutions to this exist, the most promising being virtio-iommu, but also swiotlb and PCI ATS/PRI
+* **security**: the device has access to the entire guest physical memory, which may change the 
+  security posture of firecracker. 
+    * The device will need to be cleared before being attached to avoid cross-VM interferences.
+    * Compatibility with the secret-hiding initiative to harden Firecracker security posture needs 
+      to be carefully evaluated.
+* **snapshot/resume**: it will likely not be possible to snapshot external PCIe devices, 
+  therefore snapshot/resume will not be supported for active/online passed-through devices.
+    * support for resumption with offline device should be possible
+    * an alternative to this could be hotplugging a device after resume
+
+## Contribution Guidelines
+
+Before diving deeper into the required changes in Firecracker, it’s important to be clear on the 
+responsibility splitbetween the community contributors and the maintainers. 
+As this is a community-driven initiative, it will be responsibility of contributors to propose designs, 
+make changes, and work with the upstream rust-vmm community. 
+Maintainers of Firecracker will provide guidance, code reviews, project organization, facilitate rust-vmm 
+interactions, and automated testing of the new features.
+
+### Contributors
+
+* PCIe-specific development will happen on a separate feature branch `features/pcie` which maintainers will setup, 
+  with all the required CI artifacts and infrastructure.
+* Code refactors to enable PCI features should be split in a refactor merged into main and a PCI-specific part 
+  merged into the feature branch. 
+  For example, we need to rework FC device management to support PCI, the development will need to be done in main,
+  and then merged to the PCIe feature branch.
+* Generic code that is not specific to Firecracker should be discussed with the upstream rust-vmm community, and, 
+  if possible, merged in rust-vmm, unless explicit exemption is granted by the maintainers.
+* Contributors should provide design documents in case of features spanning multiple PRs to receive
+  early guidance from maintainers.
+* Contributors should not leave open PRs stale for more than two weeks.
```

Comment:
Will the maintainers commit to not let the PRs go stale from their end?

Reply:
Should've just read ahead 😄

Reply:
sure

Reply:
we do.

> Maintainers will review new PRs to the feature branch within one week.

of course we are peer in this relationship that is why we took the commitment to answer within 1 week (half of the time respect to contributor) because it is our duty set the example and guarantee a nice developer experience.